### DISCUSSION TITLE
Curiosity

### DISCUSSION THREAD

**Discussion by Vishal-thakur-01** [keycloak/keycloak#12932] [Q&A]
Actually i have recently started learning java and i want to strt contributing to this open source as soon as possible but i am not able to understand how to do it. Can you help me with that. And i am lil curious to watch and all the new and cool stuffs and how these things are done. Hoping to get some real life experience.

**Comment by abstractj** [comment]
@Vishal-thakur-01 please refer to our contribution guidelines, is the best way to learn how to contribute https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md. If you are looking about how to learn Java, I'd recommend to search on Google.
GitHub's discussions is exclusively used to discuss topics about Keycloak.

**Reply by Vishal-thakur-01** [reply]
It will help me in leaning new things right ? Actually i don't know that i will be able to contribute in this open source but i really want to see and learn all the new things, it just get me excited.

**Comment by kaocher82** [comment]
> Actually i have recently started learning java and i want to strt contributing to this open source as soon as possible but i am not able to understand how to do it. Can you help me with that. And i am lil curious to watch and all the new and cool stuffs and how these things are done. Hoping to get some real life experience.

Do you know html css JavaScript how it's work on Web browser

**Reply by Vishal-thakur-01** [reply]
No

**Comment by abstractj** [comment]
Please feel free to reach out Java, JavaScript or HTML forums. GitHub discussions is used exclusively to discuss topics related with the usage and development of Keycloak.
